sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/RSAKeyVault.cs

```diff
@@ -0,0 +1,160 @@
+        {
+            CheckDisposed();
+
+            // Verify can be done locally using the public key
+            return _publicKey.VerifyHash(hash, signature, hashAlgorithm, padding);
+        }
+
+        /// <inheritdoc/>
+        protected override byte[] HashData(byte[] data, int offset, int count, HashAlgorithmName hashAlgorithm)
+        {
+            CheckDisposed();
+
+            using (var digestAlgorithm = Create(hashAlgorithm))
+            {
+                return digestAlgorithm.ComputeHash(data, offset, count);
+            }
+        }
+
+        /// <inheritdoc/>
+        protected override byte[] HashData(Stream data, HashAlgorithmName hashAlgorithm)
+        {
+            CheckDisposed();
+
+            using (var digestAlgorithm = Create(hashAlgorithm))
+            {
+                return digestAlgorithm.ComputeHash(data);
+            }
+        }
+
+        /// <inheritdoc/>
+        public override byte[] Decrypt(byte[] data, RSAEncryptionPadding padding)
+        {
+            CheckDisposed();
+
+            try
+            {
+                return _context.DecryptData(data, padding);
+            }
+            catch (Exception e)
+            {
+                throw new CryptographicException("Error calling Key Vault", e);
+            }
+        }
+
+        /// <inheritdoc/>
+        public override byte[] Encrypt(byte[] data, RSAEncryptionPadding padding)
+        {
+            CheckDisposed();
+
+            return _publicKey.Encrypt(data, padding);
+        }
+
+        /// <inheritdoc/>
+        public override RSAParameters ExportParameters(bool includePrivateParameters)
+        {
+            CheckDisposed();
+
+            if (includePrivateParameters)
+                throw new CryptographicException("Private keys cannot be exported by this provider");
+
+            return _publicKey.ExportParameters(includePrivateParameters);
+        }
+
+        /// <inheritdoc/>
+        public override void ImportParameters(RSAParameters parameters)
+        {
+            throw new NotSupportedException();
+        }
+
+        private void CheckDisposed()
+        {
+            if (_publicKey == null)
+                throw new ObjectDisposedException($"{nameof(RSAKeyVault)} is disposed");
+        }
+
+        /// <inheritdoc/>
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _publicKey?.Dispose();
+                _publicKey = null;
+            }
+
+            base.Dispose(disposing);
+        }
+
+        private static HashAlgorithm Create(HashAlgorithmName algorithm)
+        {
+            if (algorithm == HashAlgorithmName.SHA1)
+#pragma warning disable CA5350 // Do Not Use Weak Cryptographic Algorithms
+                return SHA1.Create();
+#pragma warning restore CA5350 // Do Not Use Weak Cryptographic Algorithms
```

Comment:
For RSA, Key Vault doesn't support SHA1, IIRC. We don't want this to be some general-purpose library and should correspond only to what Key Vault supports. So I would throw an `ArgumentException` in this case.